### PR TITLE
OASIS-14: Change configprovider component to converter component

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?
This is a follow up to https://github.com/DataDog/datadog-agent/pull/26353/files. In this version, the CollectorSettings struct does not accept a ConfigProvider anymore, hence the change to using a converter to enhance the config.

### CODE REVIEW

File: comp/otelcol/converter/impl/converter.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// currently only supports a single URI in the uris slice, and this URI needs to be a file path.
+func NewConverter() (converter.Component, error) {
+	return &ddConverter{
+		confDump: confDump{
+			provided: "not supported",
+			enhanced: "not supported",
+		},
+	}, nil
+}
+
```
Comment (mx-psi) [inline comp/otelcol/converter/impl/converter.go:34]: nit: Assert that you are implementing the `confmap.Converter` interface for better errors
```suggestion

var _ confmap.Converter = (*ddConverter)(nil)

```

File: comp/otelcol/converter/impl/converter.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+func (c *ddConverter) GetProvidedConf() string {
+	return c.confDump.provided
+}
+
+// GetEnhancedConf returns a string representing the enhanced collector configuration.
+// Note: this is currently not supported.
+func (c *ddConverter) GetEnhancedConf() string {
+	return c.confDump.enhanced
+}
+
```
Comment (mx-psi) [inline comp/otelcol/converter/impl/converter.go:76]: What are these used for?

Comment (mackjmr) [reply comp/otelcol/converter/impl/converter.go:78]: The will be called by the flare extension to retrieve the provided/ enhanced configs to include in flare.

**Review by FlorentClarret — APPROVED** [review]
All good for ADT

**Review by amenasria — APPROVED** [review]
LGTM for Agent CI Exp owned file !

**Review by CelianR — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components

**Review by mx-psi — APPROVED** [review]

**Pull request merged**